File: cpp/tebakAngka.cpp
#include <iostream>
// #include <UnityEngine.UI>
// #include <TMPro>
using namespace std;

int main() {
    // diluar Start dan Update
    int soal;
    int playerInput;

    // ini adalah Start
    srand(time(NULL));
    soal = rand() % 1000 + 1;

    // ini adalah Update
    while (1)
    {
        cout << "Input Tebakan Anda\t: ";

        cin >> playerInput;

        if (playerInput < soal) {
            cout << "Tebakan kurang\n";
        }
        else if (playerInput > soal) {
            cout << "Tebakan lebih\n";
        }
        else {
            cout << "Benarr!\n";
            break;
        }
    }

    return 0;
}

File: Assets/TebakAngka.cs
// #include <iostream>
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

using UnityEngine.UI;
using TMPro;

public class TebakAngka : MonoBehaviour
{   
    int soal;
    int playerInput;

    public TMP_InputField playerInputField;
    public TextMeshProUGUI hasilText;


    void Start()
    {
        soal = Random.Range(0, 1000);
    }



    void Update()
    {
        // cout << soal;
        Debug.Log(soal);

        //  cin >> playerInput;
        int.TryParse(playerInputField.text, out playerInput);

        // if (playerInput < soal)
        if (playerInput < soal) {
            // cout << "Tebakan kurang\n";
            Debug.Log("Tebakan kurang");
            hasilText.text = "Tebakan kurang";
        }
        // else if (playerInput > soal)
        else if (playerInput > soal) {
            // cout << "Tebakan lebih\n";
            Debug.Log("Tebakan lebih");
            hasilText.text = "Tebakan lebih";
        }
        // else
        else {
            // cout << "Benarr!\n";
            Debug.Log("Benarr!");
            hasilText.text = "Benarr!";
            Application.Quit();
        }
    }
}
